src/games/hostgamewidget.py

```diff
@@ -86,19 +101,56 @@ def hosting(self):
             # TODO: Feedback to the UI that the name must not be blank.
             return
 
+        self.friends_only = self.radioFriends.isChecked()
         if self.passCheck.isChecked():
-            password = self.passEdit.text()
+            self.password = self.passEdit.text()
         else:
-            password = None
+            self.password = None
+        self.save_last_hosted_settings()
+
+
+        self.save_last_hosted_settings()
+
+
+        # Make sure the binaries are all up to date, and abort if the update fails or is cancelled.
+        if not fa.check.game(self.parent.client):
+            return
+
+        # Ensure all mods are up-to-date, and abort if the update process fails.
+        if not fa.check.check(self.featured_mod):
+            return
+        if self.iscoop and not fa.check.map(self.mapname, force=True):
+            return
+
+        modnames = [str(moditem.text()) for moditem in self.modList.selectedItems()]
+        mods = [self.mods[modstr] for modstr in modnames]
+        modvault.setActiveMods(mods, True)
+
+        self.parent.client.host_game(title=self.title,
+                                 mod=self.featured_mod,
+                                 visibility="friends" if self.friends_only else "public",
+                                 mapname=self.mapname,
+                                 password=self.password)
 
-        # TODO: Remove this ridiculous use of a parent pointer.
-        map = self.parent.gamemap
 
-        self.parent.save_last_hosted_settings(name, map, password, self.radioFriends.isChecked())
         self.done(1)
+        return
+
 
     def mapChanged(self, index):
-        self.parent.gamemap = self.mapList.itemData(index)
+        self.mapname = self.mapList.itemData(index)
 
-        self.message['mapname'] = self.parent.gamemap
+        self.message['mapname'] = self.mapname
         self.game.update(self.message, self.parent.client)
+
+    def save_last_hosted_settings(self):
```

Comment:
Looks like this method still exists on `GamesWidget`, is that intended?

Would like to see these migrated to use the persisted properties sometime in the future, out of scope of this PR though :)